src/loadlib.js

```diff
@@ -36,21 +37,41 @@ const AUXMARK       = [1];
 */
 let lsys_load;
 if (WEB) {
-    lsys_load = function(L, path) {
+    lsys_load = function(L, path, seeglb) {
+        path = lua.to_jsstring(path);
         let xhr = new XMLHttpRequest();
-        xhr.open("GET", lua.to_jsstring(path), false);
+        xhr.open("GET", path, false);
         xhr.send();
 
-        if (xhr.status >= 200 && xhr.status <= 299) {
-            return eval(xhr.response);
-        } else {
+        if (xhr.status < 200 || xhr.status >= 300) {
             lua.lua_pushstring(L, lua.to_luastring(`${xhr.status}: ${xhr.statusText}`));
             return null;
         }
+
+        let code = xhr.response;
+        /* Add sourceURL comment to get path in debugger+tracebacks */
+        if (!/\/\/[#@] sourceURL=/.test(code))
```

Comment:
Should this test to see if sourceURL is on the final line of the file?

Though shouldn't matter too much if it doesn't work/is conservative.